payu/models/cesm_cmeps.py

```diff
@@ -313,6 +378,21 @@ def get(self, section, variable, value=None):
         else:
             return value
 
+    def get_component_list(self, value=None):
+        """
+        Get the `component_list`
+        """
+        m = re.search(
+            r'component_list:\s*(.*)',
+            self.contents
+        )
+
+        if m is not None:
+            components_str = m.group(1).strip()
+            return components_str.split()
+        else:
+            return value
```

Comment:
Why does this return an input variable? It isn't used this way in the code.

Reply:
I just followed the example of the `get()` function

Reply:
Oh right. Unless you think it is likely this functionality will be used at some point in the near future I'd recommend removing it. It confuses the intent, and if you need proof, I was confused.

Extraneous code is still code that must be tested and maintained.

So I'd say remove the input variable and just return `None`. Or you could do ultra minimalist:
```python
    def get_component_list(self):
        """
        Get the `component_list`
        """
        m = re.search(
            r'component_list:\s*(.*)',
            self.contents
        )
        return None if m is None else m.group(1).strip().split()
```

Or probably better, check you're getting what you expect even if not `None`
```python
    def get_component_list(self):
        """
        Get the `component_list`
        """
        m = re.search(
            r'component_list:\s*(.*)',
            self.contents
        )

        try:
            return m.group(1).strip().split()
        except SomeExceptionsThatAreOkToIgnore:
            return None
```

Reply:
ok done